lib/runtime/goto.js

```diff
@@ -2,20 +2,55 @@
 
 import path from 'path'
 import fs from 'fs'
+import { CompositeDisposable } from 'atom'
 
 import { client } from '../connection'
 import modules from './modules'
 import { isValidScopeToInspect } from '../misc/scopes'
 import { getWord, getWordRangeAtBufferPosition, isValidWordToInspect } from '../misc/words'
+import { getLocalContext } from '../misc/blocks'
 
-const methods = client.import('methods')
+const goto = client.import('goto')
 
+const grammar = atom.grammars.grammarForScopeName('source.julia')
 const includeRegex = /(include|include_dependency)\(".+\.jl"\)/
 const filePathRegex = /".+\.jl"/
 
 class Goto {
   activate (ink) {
     this.ink = ink
+    this.subscriptions = new CompositeDisposable()
+    this.refreshFiles = new Set()
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors(editor => {
+        if (
+          !this.refreshFiles.has(editor.getPath()) &&
+          editor.getGrammar() === grammar
```

Comment:
I've used `ed.getGrammar().id === 'source.julia'` checks for this, might be worth making that change.

Reply:
Okay, will use that instead.